internal/clients/producer.go

```diff
@@ -35,8 +36,10 @@ func InitProducer(url string, clientId string, tlsConfig *tls.Config) (sarama.Sy
 	config.Producer.Retry.Max = 5
 	config.Producer.RequiredAcks = sarama.WaitForAll // bit of an opinion here, tenant req. could vary?
 	config.Producer.Return.Successes = true
-	config.Net.TLS.Enable = true
-	config.Net.TLS.Config = tlsConfig
+	if tlsConfig != nil {
```

Comment:
This code is not affected and works as designed, but in the name of exposing you to all the Go features I can while you pick it up, this is a good place to point out the caveats of zero values if you've not seen it yet:
https://codefibers.pl/blog/golang-why-nil-is-not-always-nil

Reply:
Great article!! Thanks! I guess that in our case we don't have a problem anyway, controlling the code we know it's not going to get an empty interface.